Sources/Runes.swift

```diff
@@ -1,93 +1,75 @@
+precedencegroup AlternativePrecedence {
+    associativity: left
+    lowerThan: ComparisonPrecedence
+}
+
+precedencegroup MonadicPrecedenceRight {
+    associativity: right
+    lowerThan: LogicalDisjunctionPrecedence
+    higherThan: AssignmentPrecedence
+}
+
+precedencegroup MonadicPrecedenceLeft {
+    associativity: left
+    lowerThan: LogicalDisjunctionPrecedence
+    higherThan: AssignmentPrecedence
+}
+
 /**
 map a function over a value with context
 
 Expected function type: `(a -> b) -> f a -> f b`
 Haskell `infixl 4`
 */
-infix operator <^> {
-    associativity left
-
-    // Same precedence as the equality operator (`==`)
-    precedence 130
-}
+infix operator <^>: ComparisonPrecedence
```

Comment:
Playing with this in Argo, this actually doesn't work. `ComparisonPrecedence` doesn't have an associativity so you can't actually chain these together. Fun stuff.